rskj-core/src/main/java/co/rsk/net/sync/SnapshotPeersInformation.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.net.sync;
+
+import co.rsk.net.Peer;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This is mostly a workaround because SyncProcessor needs to access Peer instances.
+ * TODO(mc) remove this after the logical node abstraction is created, since it will wrap
```

Comment:
do we really need this interface? Can't we simply add a few extra methods to the `PeersInformation` class?

Reply:
Please refer to [this comment](https://github.com/rsksmart/rskj/pull/2504#discussion_r1647625917) for further details

Reply:
> This is mostly a workaround because SyncProcessor needs to access Peer instances.

could you pls elaborate a bit more on this javaadoc? don't quite get it.

looks like `SyncPeerStatus getOrRegisterPeer(Peer peer)` shouldn't be in this interface?

Reply:
We do need it in the interface, we use `SnapshotProcessor` which interacts with `SnapshotPeersInformation` **instead** of `PeersInformation` for a specific purpose: to control how peer information is accessed. [Check here](https://github.com/rsksmart/rskj/pull/2504/files#diff-03b3e22c54ead80f02d7379f0a383e92313300d1e68b2abf6a412fb834d4641eR70). By using `SnapshotPeersInformation`, we ensure that only methods related to snapshot peers are available and usable. This design decision helps manage and restrict access to peer-related operations within the snapshot context.`Snapshotprocessor` uses `getOrRegisterPeer(Peer peer)`, so we need to keep it there, and it uses the default method declared in _`Peersinformation`_:
```
public SyncPeerStatus getOrRegisterPeer(Peer peer) {
        SyncPeerStatus peerStatus = this.peerStatuses.get(peer);

        if (peerStatus != null && peerNotExpired(peerStatus)) {
            return peerStatus;
        }

        return this.registerPeer(peer);
    }
```